lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -35,15 +37,39 @@ import { replaceWindowsNetworkDriveLetter } from '../../os/windows-network-drive
 import {
 	ChangeButton,
 	DetailsText,
-	Footer,
 	StepButton,
 	StepNameButton,
 	StepSelection,
-	Underline,
 } from '../../styled-components';
+import { colors } from '../../theme';
 import { middleEllipsis } from '../../utils/middle-ellipsis';
 import { SVGIcon } from '../svg-icon/svg-icon';
 
+const recentUrlImagesKey = 'recentUrlImages';
+
+function normalizeRecentUrlImages(urls: string[]) {
```

Comment:
Please add the array check, as we don't know what we might get from localStorage.
```typescript
function normalizeRecentUrlImages(urls: any): string[] {
	if (!Array.isArray(urls)) {
		urls = [];
	}
	return _.takeRight(urls.filter(_.isString), 5);
}
```